### ISSUE TITLE
Investigate if we can use existing canonical-envoy rock for KFP

### ISSUE THREAD

**Issue by kimwnasptd** [canonical/pipelines-rocks#155]
### Context

There is already an existing Envoy rock that was developed by the K8s team
https://github.com/canonical/envoy-rock

That rock is based on `1.28.2`, while upstream [KFP 2.3](https://github.com/kubeflow/manifests/blob/v1.9.1-branch/apps/pipeline/upstream/base/metadata/base/kustomization.yaml#L13) is using Envoy `1.27.3`
https://github.com/kubeflow/pipelines/blob/2.3.0/third_party/metadata_envoy/Dockerfile#L15

The Pipelines envoy image is just upstream envoy plus some license files. So we should investigate if we could use the Envoy rock from K8s team instead of maintaining our own.

### What needs to get done

1. Figure out where this envoy rock is published
2. Try to use the published envoy rock instead of upstream metadata-envoy image
3. Connect to the KFP UI and see if there will be any errors
4. Discuss with K8s team
    1. Where are they using this rock
    2. Who owns it (is it still them)? 

### Definition of Done

1. We have an understanding of whether we can use this rock, or if not a list of issues
2. We have a clear understanding of who owns this rock and where else it's used

**Comment by orfeas-k** [comment]
### TL;DR
The https://github.com/canonical/envoy-rock can be used without any issues in the envoy charm :tada: . Before integrating, let's first add better integration tests (https://github.com/canonical/envoy-operator/issues/106).

### Changes needed
1. Integrate the image `ghcr.io/canonical/envoy:1.28.2`.
3. Modify the command since the binary [is placed in a different directory](https://github.com/canonical/envoy-rock/blob/main/1.28.2/rockcraft.yaml#L19) than the upstream's image
```diff
-  "command": ("/usr/local/bin/envoy" " -c" f" {config_path}"),
+  "command": ("envoy" " -c" f" {config_path}"),
```
Both changes can be seen in https://github.com/canonical/envoy-operator/pull/144.

### Difference in versions
We 've seen in the past that different versions may have a different configuration schema. Right now, upstream kfp uses 1.27 and we didn't hit such issue when using the mentioned 1.28 rock. However, we should keep this in mind for the future and have thorough integration tests that will spot malfunctioning envoy.

### Current rock state
Talking to the k8s team, this is what they replied about the rock's state.

> the k8s team created this rock and in that since we still own it, but the future plans are TBD and will depend on customer priorities. I'm not aware of it being used anywhere yet. The work planned is done though, so I think it's safe to think of it as stable.

## Future work
1. Add tests as mentioned in https://github.com/canonical/envoy-operator/issues/106.
2. Merge https://github.com/canonical/envoy-operator/pull/144

**Comment by NohaIhab** [comment]
thanks @orfeas-k! As you mentioned the next step in this effort is https://github.com/canonical/envoy-operator/issues/106
